powermon/protocols/pi30.py

```diff
@@ -826,6 +786,48 @@
     },
 }
 
+NEW_FORMAT_COMMANDS = {
+    "QID": {
+        "name": "QID",
+        "description": "Device Serial Number inquiry",
+        "help": " -- queries the device serial number",
+        "result_type": ResultType.SINGLE,
+        "reading_definitions": [{"description": "Serial Number", "reading_type": ReadingType.MESSAGE, "response_type": ResponseType.STRING}],
+        "test_responses": [b"(9293333010501\xbb\x07\r"],
+    },
+    "Q1": {
+        "name": "Q1",
+        "description": "Q1 query",
+        "result_type": ResultType.INDEXED,
+        "reading_definitions": [
+            {"index":0, "reading_type":ReadingType.TIME, "response_type":ResponseType.INT, "description":"Time until the end of absorb charging"},
```

Comment:
I was going around in circles on response_type and reading_type - this looks like the correct 'landing', ie response_type is limited to what the data is
response_type will need to include the struct.unpack types when we get there (BigHextoInt etc) but Im liking where youve got to

Reply:
I'm glad you like it, I think it's turned out to be pretty readable. Only part that gets me sometimes is that Result, Reading, and Response are all too similar in spelling when I'm reading quickly.

Reply:
yes, and we could argue in circle on which is which for ever, eg I still see response as what the device sends (currently result though i dont know what the smaller response piece should be in this way of thinking (full response and response item but thats too many letters..)
Anyway current way works, just need to keep it consistent...